ext/reflection/php_reflection.c

```diff
@@ -5715,6 +5715,106 @@ ZEND_METHOD(reflection_property, hasType)
 }
 /* }}} */
 
+/* {{{ proto public bool ReflectionProperty::hasDefaultValue()
+   Returns whether property has a default value */
+ZEND_METHOD(reflection_property, hasDefaultValue)
+{
+	reflection_object *intern;
+	property_reference *ref;
+	zend_property_info *prop_info;
+	zval *prop, prop_copy;
+	zend_class_entry *ce;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	GET_REFLECTION_OBJECT_PTR(ref);
+
+	prop_info = ref->prop;
+	ce = prop_info->ce;
+
+	prop = NULL;
+	if ((prop_info->flags & ZEND_ACC_STATIC) != 0) {
+		prop = &ce->default_static_members_table[prop_info->offset];
+		ZVAL_DEINDIRECT(prop);
+	} else if ((prop_info->flags & ZEND_ACC_STATIC) == 0) {
+		prop = &ce->default_properties_table[OBJ_PROP_TO_NUM(prop_info->offset)];
+	}
+
+	if (!prop || (ZEND_TYPE_IS_SET(prop_info->type) && Z_ISUNDEF_P(prop))) {
+		RETURN_FALSE;
+	}
+
+	/* copy: enforce read only access */
+	ZVAL_DEREF(prop);
+	ZVAL_COPY_OR_DUP(&prop_copy, prop);
+
+	/* this is necessary to make it able to work with default array
+	* properties, returned to user */
+	if (Z_TYPE(prop_copy) == IS_CONSTANT_AST) {
+		if (UNEXPECTED(zval_update_constant_ex(&prop_copy, ce) != SUCCESS)) {
+			RETURN_FALSE;
+		}
+	}
+
+	if (Z_TYPE_P(prop) != IS_NULL) {
+		RETURN_TRUE;
+	}
+
+	RETURN_FALSE;
+}
+/* }}} */
+
+/* {{{ proto public bool ReflectionProperty::hasDefaultValue()
+   Returns the default value of a property */
+ZEND_METHOD(reflection_property, getDefaultValue)
+{
+	reflection_object *intern;
+	property_reference *ref;
+	zend_property_info *prop_info;
+	zval *prop, prop_copy;
+	zend_class_entry *ce;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	GET_REFLECTION_OBJECT_PTR(ref);
+
+	prop_info = ref->prop;
+	ce = prop_info->ce;
+
+	prop = NULL;
+	if ((prop_info->flags & ZEND_ACC_STATIC) != 0) {
+		prop = &ce->default_static_members_table[prop_info->offset];
+		ZVAL_DEINDIRECT(prop);
+	} else if ((prop_info->flags & ZEND_ACC_STATIC) == 0) {
+		prop = &ce->default_properties_table[OBJ_PROP_TO_NUM(prop_info->offset)];
+	}
+
+	if (!prop || (ZEND_TYPE_IS_SET(prop_info->type) && Z_ISUNDEF_P(prop))) {
+		return;
```

Comment:
So we silently return null if no default? I guess  this is consistent with existing behavior for things like getReturnType(), which do the same...

Reply:
I believe it makes sense, the other approach could be an exception, but that might be a bit harsh  in many cases requiring additional hasDefaultType checks.

Reply:
Yeah, I think it's fine as is.